components/automate-ui/src/app/pages/user-details/user-details.component.html

```diff
@@ -29,29 +29,38 @@
                   <span class="label">Full Name <span aria-hidden="true">*</span></span>
                   <chef-input ngDefaultControl formControlName="fullName"></chef-input>
                 </label>
-                <chef-error *ngIf="(editForm.get('fullName').hasError('required') || editForm.get('fullName').hasError('pattern')) && editForm.get('fullName').dirty">
+                <chef-error
+                  *ngIf="(editForm.get('fullName').hasError('required') || editForm.get('fullName').hasError('pattern')) && editForm.get('fullName').dirty">
                   Full Name is required.
                 </chef-error>
               </chef-form-field>
             </form>
           </ng-container>
           <div class="container-big-header inline-container">
-            <span class="text"> {{ user.id }}</span>
+            <span class="text"> {{ user?.id }}</span>
           </div>
         </div>
         <div class="button-column">
           <ng-container *ngIf="!editMode">
-            <chef-button secondary class="edit-button" (click)="setEditMode(true)">
-              <chef-icon>mode_edit</chef-icon>
-              <span>Edit</span>
-            </chef-button>
-            <chef-button *ngIf="isAdminView" secondary caution class="delete-button" (click)="openDeleteConfirmationModal()">
-              <chef-icon>delete</chef-icon>
-              <span>Delete user</span>
-            </chef-button>
+            <app-authorized [allOf]="[['/auth/users/{username}', 'put', user?.id], ['/iam/v2beta/users/{username}', 'put', user?.id],
```

Comment:
the editing capabilities of this page really fall apart if you don't have GET permission, so I decided to add an authorization check around the `Edit` and `Delete` buttons

Reply:
Thank you for adding that!!